Create GPS Subsystem.md

JoshuaEgwuatu opened the pull request.

JoshuaEgwuatu committed:
Create GPS Subsystem.md

JesseTNRoberts reviewed (commented):
These constraints aren't precise enough to be design criteria

fnalheji42 committed:
Update GPS Subsystem.md

updated the constraints + added more resources

JesseTNRoberts commented on Environmental Sensing/GPS Subsystem.md:
```diff
@@ -0,0 +1,219 @@
+# Function of the subsystem:
```
Why are we discussing component selection here?

JesseTNRoberts commented on Environmental Sensing/GPS Subsystem.md:
```diff
@@ -0,0 +1,219 @@
+
+The GPS subsystem works with the top-level controller and robotic operating system (ROS). The controller manages different parts of the robot and can use GPS data along with other sensors to make decisions and control its actions. ROS helps the GPS subsystem connect with other systems like mapping and path planning, making everything work together smoothly.
+
+## Function:
```
This is way too long for a description of what this system is to do. Be brief here.

JesseTNRoberts commented on Environmental Sensing/GPS Subsystem.md:
```diff
@@ -0,0 +1,219 @@
+
+
+
+| No.|  Specifications and Constraints                                                                                             | 
```
These aren't precise. There is not way to design to these constraints. Further, there are constraints that haven't been identified here that are appropriate.

JesseTNRoberts reviewed (commented):
There's no way to design or analyze without first having a firm idea of what the constraints are.

fnalheji42 committed:
Update GPS Subsystem.md

function of the subsystem is updated 
function is updated 
specification and constraints is updated

JesseTNRoberts reviewed (commented):
Did not review. Previous comments have not been addressed.

fnalheji42 committed:
Update GPS Subsystem.md

Function is updated
Constraints is updated 
Analysis is updated

JesseTNRoberts commented on Environmental Sensing/GPS Subsystem.md:
```diff
@@ -0,0 +1,247 @@
+# Function of the subsystem:
+
+By incorporating a GPS and compass device into the Autonomous Crawlspace Inspection Robot, it would have the capability to accurately determine its position within the crawlspace. This integration of GPS would enable the robot to generate a map of the crawlspace, displaying its current location and the areas it has explored. By combining GPS and compass data, the robot can plan efficient inspecti ...
```
I can't review this without understanding what this subsystem is supposed to do.

Tell me only what functionality you expect this to have. Don't write a paragraph. Write a single sentence that says what this system's purpose is. What do you expect it to do?

JesseTNRoberts reviewed (commented):
More words aren't necessarily better. Say exactly enough to get your point across and no more.

Your analysis should address exactly the system constraints. The best thing to do is to number the constraints and address them in the same order.

fnalheji42 committed:
Update GPS Subsystem.md

Function is updated 
constraints are updated

JesseTNRoberts commented on Environmental Sensing/GPS Subsystem.md:
```diff
@@ -0,0 +1,232 @@
+   - Speed accuracy: 0.006 m/sec or less 
+
+
+## Constraint 3: 
```
This isn't a constraint

JesseTNRoberts commented on Environmental Sensing/GPS Subsystem.md:
```diff
@@ -0,0 +1,232 @@
+
+
+
+# Analysis 
```
Analysis comes after the design. Reorder.

JesseTNRoberts reviewed (commented):
There are many things in this signoff that need to be trimmed. Make sure it is the proper order. The schematic is not correct and needs more work. Specifically, this needs to be integrated into the larger circuit for the bot.

fnalheji42 committed:
Update GPS Subsystem.md

constraint 3 is updated 
Analysis is updated 
Schematics are updated

JesseTNRoberts reviewed (commented):
Probably need to reconsider the use of the GPS device sense we won't get signal. From discussion after class.

fnalheji42 committed:
Update GPS Subsystem.md

JesseTNRoberts reviewed (commented):
I think that needing 3 repeaters make this to unwieldy to expect someone to use. Again, let's look at a magnetic compass solution.

fnalheji42 committed:
Update GPS Subsystem.md

JesseTNRoberts commented on Environmental Sensing/GPS Subsystem.md:
```diff
@@ -0,0 +1,260 @@
+# Function of the subsystem:
+
+The GPS system is expected to provide real-time location tracking of the autonomous crawlspace inspection robot to enhance the accuracy of environmental data collection.
```
This isn't the desired function we discussed.

JesseTNRoberts commented on Environmental Sensing/GPS Subsystem.md:
```diff
@@ -0,0 +1,260 @@
+# Constraints:
+
+## Constraint 1: 
+The robot shall be capable of acquiring a reliable GPS signal within a time limit of less than 0.06 seconds, considering limited sky visibility and potential signal blockage.
```
Why?

JesseTNRoberts reviewed (commented):
We discussed in class that using a GPS was not a good option. We have been having the same conversation for 3 weeks now. You must propose a design that may function. You correctly established very early on that GPS would not work. So, I strongly suggest that you update the intended function (the objective is not real time location) and propose a solution that does not rely on GPS.

fnalheji42 committed:
Update GPS Subsystem.md

fnalheji42 committed:
Update GPS Subsystem.md

JesseTNRoberts commented on Environmental Sensing/GPS Subsystem.md:
```diff
@@ -0,0 +1,142 @@
+# Function of the subsystem:
+Using the compass enables the robot to determine its orientation and optimize path planning.
```
How does it permit path planning optimization? 

Before beginning any design, be absolutely certain that you understand the purpose. The compass has no use for path planning in this context. 

The purpose we have discussed for this system is to make it possible to overlay the map generated by the existing robot software onto a satellite map of the area. You have established that GPS can't get signal and won't be viable here. Good. So, how can it be done?

Without knowing what the system is precisely intended to do, it is impossible to develop constraints for the system. Without constraints, it is impossible to design the system.

fnalheji42 replied:
I have rewritten the function, but  I have not yet posted the updated function on GitHub

The compass subsystem continuously reads and processes data to calculate the robot's heading, enabling navigation, mapping, and overlaying the robot's map onto a satellite map in the autonomous crawlspace inspection robot.

Additionally, the constraints for this subsystem include considerations related to magnetic interference, calibration, and precision limitations.

I hope this meets the expectations of this subsystem, please let me know if this is ok

fnalheji42 committed:
Update GPS Subsystem.md

JesseTNRoberts commented:
The only change is that I want to use a cheaper microcontroller. There's no need for an atmega.
